File: android/app/src/debug/java/com/laneshadow/sandbox/stories/molecules/LSToolbarStory.kt
package com.laneshadow.sandbox.stories.molecules

import androidx.compose.runtime.Composable
import com.laneshadow.theme.generated.LaneShadowTheme.IconName
import com.laneshadow.ui.molecules.LSToolbar
import com.laneshadow.ui.molecules.LSToolbarAction
import com.laneshadow.ui.molecules.LSToolbarLeading
import com.laneshadow.ui.molecules.LSToolbarTrailing
import com.nativesandbox.model.ComponentTier
import com.nativesandbox.model.Story

object LSToolbarStory {
    val all: List<Story> = listOf(
        story(
            id = "molecules.toolbar.backTitleAction",
            name = "Back + Title + Action",
            summary = "Back affordance, centered title, and single trailing action.",
        ) {
            LSToolbar(
                leading = LSToolbarLeading.Back(onClick = {}),
                title = "Details",
                trailing = LSToolbarTrailing.Action(icon = IconName.Menu, onClick = {}),
            )
        },
        story(
            id = "molecules.toolbar.titleOnly",
            name = "Title Only",
            summary = "Title-centered toolbar with no leading or trailing actions.",
        ) {
            LSToolbar(
                title = "Title Only",
            )
        },
        story(
            id = "molecules.toolbar.titleTwoActions",
            name = "Title + Two Actions",
            summary = "Centered title with two trailing action buttons.",
        ) {
            LSToolbar(
                title = "Library",
                trailing = LSToolbarTrailing.Actions(
                    first = LSToolbarAction(icon = IconName.Menu, onClick = {}),
                    second = LSToolbarAction(icon = IconName.Share, onClick = {}),
                ),
            )
        },
        story(
            id = "molecules.toolbar.noBack",
            name = "No Back Button",
            summary = "Single trailing action and no leading back affordance.",
        ) {
            LSToolbar(
                title = "Saved Routes",
                trailing = LSToolbarTrailing.Action(icon = IconName.Bookmark, onClick = {}),
            )
        },
    )
}

private fun story(
    id: String,
    name: String,
    summary: String,
    content: @Composable () -> Unit,
): Story =
    Story(
        id = id,
        tier = ComponentTier.Molecule,
        component = "LSToolbar",
        name = name,
        summary = summary,
        content = { MoleculeStoryFrame(content) },
    )

File: ios/LaneShadow/Sandbox/Stories/Molecules/LSToolbarStory.swift
import LaneShadowTheme
import NativeSandbox
import SwiftUI

@MainActor
enum LSToolbarStory {
    static let all: [Story] = [
        Story(
            id: "molecules.toolbar.back-title-action",
            tier: .molecule,
            component: "Toolbar",
            name: "Back+Title+Action",
            summary: "Toolbar with back leading slot, centered title, and one trailing action."
        ) { _ in
            LSToolbar(
                leading: .back(action: {}),
                title: "Details",
                trailing: .action(icon: .menu, action: {})
            )
        },
        Story(
            id: "molecules.toolbar.title-only",
            tier: .molecule,
            component: "Toolbar",
            name: "Title Only",
            summary: "Toolbar with centered title and empty leading/trailing slots."
        ) { _ in
            LSToolbar(title: "Overview")
        },
        Story(
            id: "molecules.toolbar.title-two-actions",
            tier: .molecule,
            component: "Toolbar",
            name: "Title+Two Actions",
            summary: "Toolbar with centered title and two trailing icon actions."
        ) { _ in
            LSToolbar(
                title: "Map",
                trailing: .actions([
                    LSToolbarAction(icon: .sliders, action: {}),
                    LSToolbarAction(icon: .menu, action: {}),
                ])
            )
        },
        Story(
            id: "molecules.toolbar.no-back-button",
            tier: .molecule,
            component: "Toolbar",
            name: "No Back Button",
            summary: "Toolbar with no leading back button and a trailing action."
        ) { _ in
            LSToolbar(
                leading: .none,
                title: "Saved Routes",
                trailing: .action(icon: .bookmark, action: {})
            )
        },
    ]
}
